sql/5_0_2-to-5_0_3_upgrade.sql

```diff
@@ -515,3 +515,82 @@ ALTER TABLE `api_token` ADD `token_api` varchar(40);
 #IfMissingColumn api_token patient_id
 ALTER TABLE `api_token` ADD `patient_id` bigint(20) NOT NULL;
 #EndIf
+
+#IfNotTable fhir_forced_id
+CREATE TABLE `fhir_forced_id` (
+    `pid` bigint(20) NOT NULL,
+    `forced_id` varchar(100) NOT NULL,
+    `resource_pid` bigint(20) NOT NULL,
+    `resource_type` varchar(100) DEFAULT '',
+    PRIMARY KEY (`pid`),
+    UNIQUE KEY `idx_forcedid_resid` (`resource_pid`),
+    UNIQUE KEY `idx_forcedid_type_resid` (`resource_type`,`resource_pid`),
+    KEY `idx_forcedid_type_forcedid` (`resource_type`,`forced_id`)
+) ENGINE=InnoDB;
+#EndIf
+
+#IfNotTable fhir_resource
+CREATE TABLE `fhir_resource` (
+    `res_id` bigint(20) NOT NULL,
+    `res_deleted_at` datetime DEFAULT NULL,
+    `res_version` varchar(7) DEFAULT NULL,
+    `has_tags` bit(1) NOT NULL,
+    `res_published` datetime DEFAULT NULL,
+    `res_updated` datetime DEFAULT NULL,
+    `sp_has_links` bit(1) DEFAULT NULL,
+    `hash_sha256` varchar(64) DEFAULT NULL,
+    `sp_index_status` bigint(20) DEFAULT NULL,
+    `res_language` varchar(20) DEFAULT NULL,
+    `sp_cmpstr_uniq_present` bit(1) DEFAULT NULL,
+    `sp_coords_present` bit(1) DEFAULT NULL,
+    `sp_date_present` bit(1) DEFAULT NULL,
+    `sp_number_present` bit(1) DEFAULT NULL,
+    `sp_quantity_present` bit(1) DEFAULT NULL,
+    `sp_string_present` bit(1) DEFAULT NULL,
+    `sp_token_present` bit(1) DEFAULT NULL,
+    `sp_uri_present` bit(1) DEFAULT NULL,
+    `res_profile` varchar(200) DEFAULT NULL,
+    `res_type` varchar(30) DEFAULT NULL,
+    `res_ver` bigint(20) DEFAULT NULL,
+    `forced_id_pid` bigint(20) DEFAULT NULL,
+    PRIMARY KEY (`res_id`),
+    KEY `idx_res_date` (`res_updated`),
+    KEY `idx_res_lang` (`res_type`,`res_language`),
+    KEY `idx_res_profile` (`res_profile`),
+    KEY `idx_res_type` (`res_type`),
+    KEY `idx_indexstatus` (`sp_index_status`),
+    KEY `fk_resource_forcedid` (`forced_id_pid`)
+) ENGINE=InnoDB;
+#EndIf
+
+#IfNotTable fhir_res_ver
+CREATE TABLE `fhir_res_ver` (
```

Comment:
Is OpenEMR going to support versioning or is this a tracking table for resources in other systems?

Reply:
lol, I was wondering who was going to have me eat crow on this.
Personally I don't see a reason except to make it easier to implement an administrative review process for updates etc.
I'm going to leave that up to you and @rhausam and @yashrajbothra I left version in tables in case.

Reply:
I too may "leave this exercise up to the reader" :smile:

I'll be in and out on paternity leave in about two weeks, so I won't have time to take it on.
I will get that pesky Patient FHIR controller sorted out though.

@yashrajbothra and @rhausam - I would recommend just getting the basic FHIR things working - POST, PUT, GET, etc for all existing OpenEMR resources before tackling versioning.

Reply:
Yes, I agree with starting with getting the basics for FHIR in OpenEMR working.  Ultimately, I expect it makes sense for the level of support for versioning in the OpenEMR FHIR resources to match whatever the underlying level of support is for it in OpenEMR generally, but we can focus on the basics first.

Reply:
> I'll be in and out on paternity leave in about two weeks, so I won't have time to take it on. 

if possible you can just review upcoming PR's. Rest all the work is on me :+1:

> I will get that pesky Patient FHIR controller sorted out though.

 Eagerly waiting to see this working :smiley:  

> I would recommend just getting the basic FHIR things working - POST, PUT, GET, etc for all existing OpenEMR resources before tackling versioning.

Yeah, This will be my primary goal in the upcoming 3 months. Although I am not sure about _history and Versioning. So, I will start with basic functionalities :)